rudder-web/src/main/scala/com/normation/rudder/web/rest/technique/TechniqueAPIService6.scala

```diff
@@ -0,0 +1,114 @@
+import com.normation.rudder.batch.AsyncDeploymentAgent
+import com.normation.rudder.batch.AutomaticStartDeployment
+import com.normation.rudder.domain.RudderLDAPConstants
+import com.normation.rudder.domain.policies.ActiveTechnique
+import com.normation.rudder.domain.policies.ChangeRequestDirectiveDiff
+import com.normation.rudder.domain.policies.DeleteDirectiveDiff
+import com.normation.rudder.domain.policies.Directive
+import com.normation.rudder.domain.policies.DirectiveId
+import com.normation.rudder.domain.policies.ModifyToDirectiveDiff
+import com.normation.rudder.domain.policies.SectionVal
+import com.normation.rudder.domain.workflows.ChangeRequestId
+import com.normation.rudder.repository.RoDirectiveRepository
+import com.normation.rudder.repository.WoDirectiveRepository
+import com.normation.rudder.services.workflows.ChangeRequestService
+import com.normation.rudder.services.workflows.WorkflowService
+import com.normation.rudder.web.model.DirectiveEditor
+import com.normation.rudder.web.rest.RestUtils
+import com.normation.rudder.web.rest.RestUtils.getActor
+import com.normation.rudder.web.rest.RestUtils.toJsonError
+import com.normation.rudder.web.rest.RestUtils.toJsonResponse
+import com.normation.rudder.web.services.DirectiveEditorService
+import com.normation.rudder.web.rest.RestExtractorService
+import com.normation.utils.Control._
+import com.normation.utils.StringUuidGenerator
+import net.liftweb.common._
+import net.liftweb.http.Req
+import net.liftweb.json.JArray
+import net.liftweb.json.JValue
+import net.liftweb.json.JsonDSL._
+import com.normation.rudder.web.rest.RestDataSerializer
+import com.normation.rudder.domain.workflows.ChangeRequestId
+import com.normation.cfclerk.domain.TechniqueId
+import com.normation.cfclerk.services.TechniqueRepository
+import com.normation.utils.Control.boxSequence
+
+case class TechniqueAPIService6 (
+    readDirective        : RoDirectiveRepository
+  , restDataSerializer   : RestDataSerializer
+  , techniqueRepository  : TechniqueRepository
+  ) extends Loggable {
+
+  def serialize(technique : Technique, directive:Directive) = restDataSerializer.serializeDirective(technique, directive, None)
+
+  def listDirectives (techniqueName : TechniqueName, versions: Option[List[TechniqueVersion]]) = {
+    def serializeDirectives (directives : Seq[Directive], techniques : Map[TechniqueVersion, Technique]) = {
+     for {
+       directive <- directives
+       if ( versions match {
+         case None => true
+         case Some(versions) => versions.contains(directive.techniqueVersion)
+       })
+     } yield {
+      techniques.get(directive.techniqueVersion) match {
+        case None => Failure(s"Version ${directive.techniqueVersion} of Technique '${techniqueName.value}' does not exists")
+        case Some(technique) => Full(serialize(technique,directive))
+      }
+     }
+    }
+
+    for {
+
+     optActiveTechnique <- readDirective.getActiveTechnique(techniqueName)
```

Comment:
I believe that using the full directive library would be better, because: 1/ you only make one call to ldap 2/ you just have to call the correct method on the fullactivetechnique lib and filter the result